src/middlewared/middlewared/plugins/iscsi.py

```diff
@@ -910,8 +852,7 @@ async def disk_choices(self, exclude):
 
         zvol_query_filters = [('type', '=', 'ZVOL')]
         for e in exclude:
-            if e:
-                zvol_query_filters.append(('path', '!=', e))
+            zvol_query_filters.append(('path', '!=', e))
```

Comment:
Since we are changing this, could probable be
`zvol_query_filters.append(('path', 'nin', exclude))`

thoughts?